cmd/interceptors/main.go

```diff
@@ -131,10 +131,12 @@ func updateCRDWithCaCert(ctx context.Context, cfg *rest.Config, caCert []byte) e
 		return err
 	}
 	for i := range clusterInterceptorList.Items {
-		if bytes.Equal(clusterInterceptorList.Items[i].Spec.ClientConfig.CaBundle, []byte{}) {
-			clusterInterceptorList.Items[i].Spec.ClientConfig.CaBundle = caCert
-			if _, err := tc.TriggersV1alpha1().ClusterInterceptors().Update(ctx, &clusterInterceptorList.Items[i], metav1.UpdateOptions{}); err != nil {
-				return err
+		if v, ok := clusterInterceptorList.Items[i].GetLabels()["part-of"]; ok && v == "core-interceptors" {
```

Comment:
Using labels work but I don't think they are a good long term solution - anyone can add/modify labels + any other interceptor that wants to reuse this logic (e.g. Andrea's buildid one) will have to add this core-interceptor label

Reply:
I believe that for custom interceptors like  `buildid` (e.g. Andrea's buildid one)  if runs as `https` then its the user responsibility to create certs and give it in CI caBundle and in those cases core-interceptor label not needed
Let me know if this assumption is not correct 

also `core-interceptor` label added to differentiate from core interceptors

Reply:
I think we can provide the same auto certificate creation feature that's available to the core interceptor depending upon some annotation/labels. It's a useful feature. This label/annotation can be added to the core-interceptor.